BrowserSelect/Program.cs

```diff
@@ -146,7 +147,43 @@ public static string ProgramFilesx86()
             return Environment.GetEnvironmentVariable("ProgramFiles");
         }
 
+        public static bool DoesAutoRuleMatch(Uri uri, string pattern)
+        {
+            try {
+                // test for whether url user is visiting matches the pa
+                if (uri.Equals(pattern))
+                {
+                    return true;
+                }
+            } catch (UriFormatException e) 
+            { 
+                // will probably get this with wildcard strings, ignore
+            }
+            // for the regex, replace every '*' with '.*' so the regex can understand it
+            // of course this means that urls with * in them won't match very well, but 
+            // assume that's not very common kind of url
+
+            // tidy up the pattern for regex
+            string newpattern = Regex.Escape(pattern);
+            // unfortunately, this means that the wildcard * that we wanted in the pattern
+            // has now been escaped to \*, so let's reverse that 
+            newpattern = newpattern.Replace(@"\*", ".*");
```

Comment:
What if the URL itself included `*`? For example `http://google.com/search?q=*` which is a google search for `*`, would match a google search for anything. There should be a way to differentiate from a literal `*` (maybe from Always button) and a wildcard (entered by the user).